### PR TITLE
fix: Don't throw NPE on null columns

### LINKED ISSUE #3617: Pull queries: Error in SELECT when ROWKEY contains the string null
**Describe the bug**
Assume a table with schema: 
`ROWTIME|ROWKEY|METRICS_NAME |CLUSTER|TOPIC_NAME|ENV|AVG_VALUE`

Assume the table contains the following two rows:
``|1571347727133 |fetch_latency_ms|+|null|+|null|+|staging : Window{start=1571347680000 end=-}|fetch_latency_ms|null|null|staging|9.44924|
|1571347727133 fetch_latency_ms|+|cluster_lmn|+|topic_2|+|staging : Window{start=1571347680000 end=-}  |fetch_latency_ms |cluster_lmn |topic_2 |staging| 7.988785``

Note how the key contains the string null in the first row. 

Issue query: 
`select * from metrics_table where rowkey='fetch_latency_ms|+|null|+|null|+|null';`
Error reported:
`Server Error`

**To Reproduce**
Steps to reproduce the behavior, include:
1. The version of KSQL.
Master
2. Sample source data.
Provided above
3. Any SQL statements you ran
Provided above

**Expected behavior**
Support rowkeys that contain the null string

**Actual behaviour**
A clear and concise description of what actually happens, including:
1. CLI output
`Server Error`
2. Error messages
`Server Error`
3. KSQL logs
ksql-cli logs contain:
`[2019-10-17 15:47:15,556] ERROR Server Error
 (io.confluent.ksql.cli.console.Console:323)`

**Additional context**

### PR DESCRIPTION
**Pull request by vpapavas**
Fixes #3617 

### Description 
`TableRowsEntityFactory.createRows` was throwing a NPE when a column of a row was null.

### Testing done 
Added extra tests to check if a column of different types is NULL.

CLI output for  query: `ksql> SELECT * FROM METRICS_TABLE WHERE ROWKEY='fetch_latency_ms|+|cluster_xyz|+|topic_3|+|null';`

**Previous** output: `Server Error`

**After** changes output: 

```
 ROWKEY STRING KEY                               | WINDOWSTART BIGINT KEY | METRICS_NAME STRING | CLUSTER STRING | TOPIC_NAME STRING | ENV STRING | AVG_VALUE DOUBLE
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
 fetch_latency_ms|+|cluster_xyz|+|topic_3|+|null | 1572304800000          | fetch_latency_ms    | cluster_xyz    | topic_3           | null       | 0.031829
---------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

CLI output for query: `ksql> SELECT * FROM ORDER_QUANTITIES where rowkey='null';`

```
 ROWKEY STRING KEY | ORDER_ID STRING | TOTAL_QTY INTEGER
---------------------------------------------------------
 null              | null          | 1
---------------------------------------------------------
```

## Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by agavra** [comment]
Thanks @vpapavas! Can you add a sample output in the CLI to make sure that everything look good? Also if you could include a table with tombstone output.

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/entity/TableRowsEntityFactory.java [hunk 1]
```diff
@@ -88,7 +88,13 @@ private static LogicalSchema addWindowFieldsIntoSchema(
       window.end().map(Instant::toEpochMilli).ifPresent(builder::add);
     });
 
-    builder.addAll(row.value().getColumns());
+    for (Object col: row.value().getColumns()) {
+      if (col == null) {
+        builder.add("null");
```
[the remark this answers is not in the document]
Comment (vpapavas) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/entity/TableRowsEntityFactory.java]: I followed the code in `GenericRow.toString()` and added different handling based on whether the value is a string or an array.

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/entity/TableRowsEntityFactory.java [hunk 2]
```diff
@@ -88,7 +89,17 @@ private static LogicalSchema addWindowFieldsIntoSchema(
       window.end().map(Instant::toEpochMilli).ifPresent(builder::add);
     });
 
-    builder.addAll(row.value().getColumns());
+    for (Object col: row.value().getColumns()) {
```
Comment (agavra) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/entity/TableRowsEntityFactory.java]: talked offline - I think a "better" fix is not to coerce to string but rather not use `ImmutableList` here, which doesn't allow nulls.

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/entity/TableRowsEntityFactoryTest.java [hunk 3]
```diff
@@ -100,6 +110,90 @@ public void shouldAddWindowedRowToValues() {
     assertThat(output.get(1), contains("y", now.toEpochMilli(), now.toEpochMilli(), false));
   }
 
+  @Test
+  public void shouldSupportNullStringColumn() {
```
Comment (agavra) [inline ksql-rest-app/src/test/java/io/confluent/ksql/rest/entity/TableRowsEntityFactoryTest.java]: I think having all these tests is a little overkill :) i'd be happy with just one test that adds null in each row:
```
    newColumns.add(null);
    newColumns.add(null);
    newColumns.add(null);
    newColumns.add(null);
```

**Pull request merged**